app/controllers/orangelight/browsables_controller.rb

```diff
@@ -1,59 +1,179 @@
 # frozen_string_literal: true
 
 class Orangelight::BrowsablesController < ApplicationController
-  # GET /orangelight/names
-  # GET /orangelight/names.json
-  def index
+  def update_facet
+    if browsing_names?
+      @facet = 'author_s'
+    elsif browsing_subjects?
+      @facet = 'subject_facet'
+    elsif browsing_titles?
+      @facet = 'name_title_browse_s'
+    end
+  end
+
+  # All of this needs to be restructured into a QueryService Class
+  def connection
+    Orangelight.browse_lists_index.connection
+  end
+
+  def odm_class
+    return CallNumberDocument if browsing_call_numbers?
+
+    BrowseListDocument
+  end
+
+  def query_response_documents(query_response)
+    query_response["response"]["docs"].map { |doc| odm_class.new(doc) }
+  end
+
+  ## The queries
+
+  def solr_default_params
+    {
+      sort: solr_default_sort,
+      fl: solr_default_fields,
+      rows: 100_000_0
+    }
+  end
+
+  def solr_model_query(model)
+    "model_s:#{model}"
+  end
+
+  def solr_default_fields
+    "id,model_s,index_i,normalized_sort,direction_s,count_i,title_s,author_s,date_s,bibid_s,holding_id_s,location_s,scheme_s"
+  end
+
+  def solr_default_sort
+    "index_i asc,normalized_sort asc" # The last-added Document should be last
+  end
+
+  # Generate the query for finding documents by model within an inclusive range of indices
+  # @return [Hash]
+  def find_range_by_model_query(model, start, last)
+    solr_query = solr_model_query(model)
+    range_parameters = {
+      q: solr_query
+    }
+    range_parameters[:fq] = "index_i:[#{start} TO #{last}]" unless start.nil? && last.nil?
+
+    solr_default_params.merge(range_parameters)
+  end
+
+  def find_range_by_model(model:, start:, last:)
```

Comment:
These queries should probably follow a `QueryService` pattern, with each query instead being offered as a method on this Class.

Reply:
I have begun implementing this on https://github.com/pulibrary/orangelight/tree/issues-1626-jrgriffiniii-browse-lists-query-service